spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/JdbcPollingChannelAdapter.java

```diff
@@ -116,6 +116,15 @@ public void setRowMapper(@Nullable RowMapper<?> rowMapper) {
 		}
 	}
 
+	/**
+	 * Set the select query.
+	 * @param selectQuery the query.
+	 * @since 5.2.1
+	 */
+	public void setSelectQuery(String selectQuery) {
+		this.selectQuery = selectQuery;
```

Comment:
`Assert.hasText(selectQuery, "'selectQuery' must be specified.");` still must be here: we can't have this component without `SELECT`